diff --git a/libavcodec/packet.h b/libavcodec/packet.h
index ca18ae6..835979e 100644
--- a/libavcodec/packet.h
+++ b/libavcodec/packet.h
@@ -21,6 +21,7 @@
 #ifndef AVCODEC_PACKET_H
 #define AVCODEC_PACKET_H
 
+#include <stdbool.h>
 #include <stddef.h>
 #include <stdint.h>
 
@@ -388,6 +389,16 @@ typedef struct AVPacket {
 
     int64_t pos;                            ///< byte position in stream, -1 if unknown
 
+    uint32_t timestamp;
+
+    uint64_t last_rtcp_ntp_time;
+
+    uint32_t last_rtcp_timestamp;
+
+    uint16_t seq;
+
+    bool synced;
+
 #if FF_API_CONVERGENCE_DURATION
     /**
      * @deprecated Same as the duration field, but as int64_t. This was required
diff --git a/libavformat/rtpdec.c b/libavformat/rtpdec.c
index fd4601e..747e0fb 100644
--- a/libavformat/rtpdec.c
+++ b/libavformat/rtpdec.c
@@ -19,6 +19,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#include <stdbool.h>
 #include "libavutil/mathematics.h"
 #include "libavutil/avstring.h"
 #include "libavutil/intreadwrite.h"
@@ -628,6 +629,7 @@ void ff_rtp_parse_set_crypto(RTPDemuxContext *s, const char *suite,
  */
 static void finalize_packet(RTPDemuxContext *s, AVPacket *pkt, uint32_t timestamp)
 {
+    bool synced = false;
     if (pkt->pts != AV_NOPTS_VALUE || pkt->dts != AV_NOPTS_VALUE)
         return; /* Timestamp already set by depacketizer */
     if (timestamp == RTP_NOTS_VALUE)
@@ -659,6 +661,22 @@ static void finalize_packet(RTPDemuxContext *s, AVPacket *pkt, uint32_t timestam
     s->timestamp = timestamp;
     pkt->pts     = s->unwrapped_timestamp + s->range_start_offset -
                    s->base_timestamp;
+
+    /* export private data (timestamps) into AVPacket */
+    if (s->last_rtcp_ntp_time != AV_NOPTS_VALUE && s->last_rtcp_timestamp) {
+        synced = true;
+        pkt->last_rtcp_ntp_time = s->last_rtcp_ntp_time;
+        pkt->last_rtcp_timestamp = s->last_rtcp_timestamp;
+    }
+    else {
+        pkt->last_rtcp_ntp_time = 0;
+        pkt->last_rtcp_timestamp = 0;
+    }
+
+    pkt->seq = s->seq;
+    pkt->timestamp = s->timestamp;
+    pkt->synced = synced;
+
 }
 
 static int rtp_parse_packet_internal(RTPDemuxContext *s, AVPacket *pkt,
diff --git a/libavformat/utils.c b/libavformat/utils.c
index 75e5350..345c1ca 100644
--- a/libavformat/utils.c
+++ b/libavformat/utils.c
@@ -1498,6 +1498,15 @@ static int parse_packet(AVFormatContext *s, AVPacket *pkt,
         out_pkt->pos          = st->parser->pos;
         out_pkt->flags       |= pkt->flags & AV_PKT_FLAG_DISCARD;
 
+        // copy over needed stuff to out_pkt
+        // which will be appended to a pkt_list and out_pkt will be
+        // reverted to default values
+        out_pkt->synced = pkt->synced;
+        out_pkt->last_rtcp_ntp_time = pkt->last_rtcp_ntp_time;
+        out_pkt->last_rtcp_timestamp = pkt->last_rtcp_timestamp;
+        out_pkt->seq = pkt->seq;
+        out_pkt->timestamp = pkt->timestamp;
+
         if (st->need_parsing == AVSTREAM_PARSE_FULL_RAW)
             out_pkt->pos = st->parser->frame_offset;
 
